### PR TITLE
Search box

### PR DESCRIPTION
**Pull request by tmccoy529**
Adds the search field for the header. Formatting should be correct now that vs code is up to date.

### CODE REVIEW

File: src/components/site/Search.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const Search = props => (
+  <form className="dg_search-container" id="search-bar">
```
Comment (martypowell) [inline src/components/site/Search.js]: can we cahnge this id to "search-form" please?

Comment (martypowell) [reply src/components/site/Search.js]: also please add a label for the input based on the w3 recommendation https://www.w3.org/WAI/tutorials/forms/labels/#hiding-the-label-element.

File: src/styles/partials/components/_search.scss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    top: $padding-small;
+    font-size: $base-font-size -2;
+    color: $black;
+    cursor: pointer;
+    background: none;
+    border: none;
+
+    i:hover {
+      color: $blue-lightest;
+      transition: all 0.3s ease 0s;
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: can you check if this is needed? if you look at _animations.scss i think this is already handled.

File: src/components/site/Search.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    <input
+      className="dg_search-text-field"
```
Comment (martypowell) [inline src/components/site/Search.js]: className here i think should be `_dg_search-input`

File: src/styles/partials/components/_search.scss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  }
+}
+.dg_search-text-field {
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: this will change, if you make the change above to `_dg_search-input`

**Comment by danfox01** [comment]
@martypowell this component raises a related issue. We are not loading Crimson Text 400i, so the label is defaulting to 600i (semibold). Is this something we can address here? Label should be Crimson 400 italic.

File: src/components/site/Search.js (hunk 1 above)
Comment (tmccoy529) [reply src/components/site/Search.js]: done

File: src/styles/partials/components/_search.scss (hunk 2 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: when this is removed the highlight does not work. So maybe it needs to be added to animations?

Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: oh sorry,  the transitions are handled. I'll remove

File: src/components/site/Search.js (hunk 3 above)
Comment (tmccoy529) [reply src/components/site/Search.js]: done

File: src/styles/partials/components/_search.scss (hunk 4 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: done

File: src/styles/partials/components/_search.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  width: 100%;
+  height: $base-spacing-unit * 10;
+  color: $black;
+
+  .dg_search-btn {
+    display: block;
+    position: absolute;
+    right: $base-spacing-unit * 2;
+    top: $padding-small;
+    font-size: $base-font-size -2;
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: font-size is broken in the browser, it doesn't understand `$base-font-size -2`. I think it looks ok without this being specified though.

File: src/styles/partials/components/_search.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
+.dg_search-container {
+  position: relative;
+  width: 100%;
+  height: $base-spacing-unit * 10;
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: you dont' need the height or the color.

File: src/styles/partials/components/_search.scss (hunk 4 above)
Comment (martypowell) [reply src/styles/partials/components/_search.scss]: It looks like you only need padding here.  
image

File: src/styles/partials/components/_search.scss (hunk 6 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: removed

File: src/styles/partials/components/_search.scss (hunk 5 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: removed the subtraction

File: src/styles/partials/components/_search.scss (hunk 4 above)
Comment (martypowell) [reply src/styles/partials/components/_search.scss]: just so its documented, font is pending convo.

File: src/styles/partials/components/_search.scss [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    right: $base-spacing-unit * 2;
+    top: $padding-small;
+    font-size: $base-font-size;
+    color: $black;
+    cursor: pointer;
+    background: none;
+    border: none;
+
+    i:hover {
+      color: $blue-lightest;
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: per @sdurphy this should be $vibrant-blue; it has not been added to the xd document.

File: src/styles/partials/components/_search.scss [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    }
+  }
+}
+.dg_search-input {
+  display: block;
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss]: per our chat, everything can be removed except the padding.

File: src/styles/partials/components/_utilities.scss [hunk 9]
```diff
@@ -0,0 +1,25 @@
+.visuallyhidden,
+h2 {
```
Comment (martypowell) [inline src/styles/partials/components/_utilities.scss]: h2 should be removed, we only want the visually hidden class.

File: src/styles/partials/components/_utilities.scss [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px;
+}
+
+.visuallyhidden.hasfocus,
+h2.hasfocus,
```
Comment (martypowell) [inline src/styles/partials/components/_utilities.scss]: this selector can be removed.

File: src/styles/partials/components/_utilities.scss [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+.visuallyhidden.focusable:active,
+footer h2.focusable:active,
```
Comment (martypowell) [inline src/styles/partials/components/_utilities.scss]: this selector can be removed.

File: src/styles/partials/components/_utilities.scss [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+.visuallyhidden.focusable:focus,
+footer h2.focusable:focus {
```
Comment (martypowell) [inline src/styles/partials/components/_utilities.scss]: this selector can be removed.

**Review by martypowell — CHANGES_REQUESTED** [review]
Per our chat `:placeholder` should be changed to use crimson italic 18px per @sdurphy and @danfox01 . This can be added to the form partial.

**Comment by martypowell** [comment]
> @martypowell this component raises a related issue. We are not loading Crimson Text 400i, so the label is defaulting to 600i (semibold). Is this something we can address here? Label should be Crimson 400 italic.

@danfox01 addressed in the most recent review.

File: src/styles/partials/components/_search.scss (hunk 7 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: changed

File: src/styles/partials/components/_search.scss (hunk 8 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss]: all removed

File: src/styles/partials/components/_utilities.scss (hunk 9 above)
Comment (tmccoy529) [reply src/styles/partials/components/_utilities.scss]: removed

File: src/styles/partials/components/_utilities.scss (hunk 10 above)
Comment (tmccoy529) [reply src/styles/partials/components/_utilities.scss]: removed

File: src/styles/partials/components/_utilities.scss (hunk 11 above)
Comment (tmccoy529) [reply src/styles/partials/components/_utilities.scss]: removed

File: src/styles/partials/components/_utilities.scss (hunk 12 above)
Comment (tmccoy529) [reply src/styles/partials/components/_utilities.scss]: removed

**Comment by tmccoy529** [comment]
@martypowell @danfox01 I added the ::placeholder class to _form.scss with the specified css properties.

File: src/styles/partials/components/_utilities.scss [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  padding: 0;
+  position: absolute;
+  width: 1px;
+}
+
+.visuallyhidden.hasfocus,
+.visuallyhidden.focusable:active,
+footer,
+.visuallyhidden.focusable:focus,
+footer {
```
Comment (martypowell) [inline src/styles/partials/components/_utilities.scss]: @tmccoy529 I'm guessing you amde these cahnges, but didn't push them up?

File: src/styles/partials/components/_form.scss [hunk 14]
```diff
@@ -1,12 +1,18 @@
...
-	}
+  label,
+  .dg_label {
+    margin-bottom: $margin-smallest;
+  }
+}
+
+::placeholder {
+  font-family: $default-body-font;
+  font-size: $small-heading-font-size;
```
Comment (martypowell) [inline src/styles/partials/components/_form.scss]: @tmccoy529 i just removed this when @sdurphy confirmed the font-size should be inherited.

File: src/styles/partials/components/_search.scss [hunk 15]
```diff
@@ -0,0 +1,23 @@
+.dg_search-container {
+  position: relative;
+  width: 100%;
+
+  .dg_search-btn {
```
Comment (martypowell) [inline src/styles/partials/components/_search.scss:5]: You can remove the following unchecked properties.

image

**Review by martypowell — CHANGES_REQUESTED** [review]
Just two quick ones.

File: src/styles/partials/components/_utilities.scss (hunk 13 above)
Comment (tmccoy529) [reply src/styles/partials/components/_utilities.scss]: must not have but I just pushed again.

File: src/styles/partials/components/_search.scss (hunk 15 above)
Comment (tmccoy529) [reply src/styles/partials/components/_search.scss:5]: deleted them

**Review by martypowell — DISMISSED** [review]

**Comment by martypowell** [comment]
@danfox01 @sdurphy this is ready for a rubber stamp

**Review by danfox01 — CHANGES_REQUESTED** [review]
Looks great with one change. Placeholder font should be Crimson Text Italic 400.

**Comment by tmccoy529** [comment]
@danfox01 the change has been made to 400, can you verify this is correct now?

**Review by danfox01 — APPROVED** [review]
Perfect

**Review by martypowell — APPROVED** [review]
Looks good thanks.

**Comment by martypowell** [comment]
@tmccoy529 this is ready to complete.

**Pull request merged**